protocol/contracts/amo/TpfAmo.sol

```diff
@@ -0,0 +1,547 @@
+ * BPTs into TEMPLE and burn them and if the price is above the TPF it will 
+ * single side deposit TEMPLE and increase its BPT position that way. Alternatively
+ */
+contract TpfAmo is Ownable {
+    using SafeERC20 for IERC20;
+
+    AMO__IBalancerVault public immutable balancerVault;
+    // @notice BPT token address
+    IERC20 public immutable bptToken;
+    // @notice Aura booster
+    AMO__IAuraBooster public immutable booster;
+    // @notice pool helper contract
+    AMO__IPoolHelper public poolHelper;
+    
+    // @notice AMO contract for staking into aura 
+    AMO__IAuraStaking public amoStaking;
+
+    address public operator;
+    IERC20 public immutable temple;
+    IERC20 public immutable stable;
+
+    // @notice lastRebalanceTimeSecs and cooldown used to control callrate 
+    // for operator
+    uint64 internal lastRebalanceTimeSecs;
+    uint64 public cooldownSecs;
+
+    // @notice balancer 50/50 pool ID.
+    bytes32 public immutable balancerPoolId;
+
+    // @notice Temple price floor denominator
+    uint256 public constant TPF_PRECISION = 10_000;
+
+    // @notice Maximum amount of tokens that can be rebalanced
+    uint256 public maxRebalanceAmount;
+
+    // @notice by how much TPF slips up or down after rebalancing. In basis points
+    uint64 public postRebalanceSlippage;
+
+    // @notice contract paused state
+    bool public paused;
+
+    // @notice temple index in balancer pool. to avoid recalculation or external calls
+    uint64 public templeBalancerPoolIndex;
+
+    event RecoveredToken(address, address, uint256);
+    event SetOperator(address);
+    event SetPostRebalanceSlippage(uint64);
+    event SetCooldown(uint64);
+    event SetRebalanceRateChange(uint256);
+    event SetPauseState(bool);
+    event StableDeposited(uint256, uint256);
+    event RebalanceUp(uint256, uint256);
+    event RebalanceDown(uint256, uint256);
+    event SetPoolHelper(address);
+    event SetMaxRebalanceAmount(uint256);
+    event WithdrawStable(uint256, uint256);
+
+    constructor(
+        address _balancerVault,
+        address _temple,
+        address _stable,
+        address _bptToken,
+        address _amoStaking,
+        address _booster,
+        uint64 _templeIndexInPool,
+        bytes32 _balancerPoolId
+    ) {
+        balancerVault = AMO__IBalancerVault(_balancerVault);
+        temple = IERC20(_temple);
+        stable = IERC20(_stable);
+        bptToken = IERC20(_bptToken);
+        amoStaking = AMO__IAuraStaking(_amoStaking);
+        booster = AMO__IAuraBooster(_booster);
+        templeBalancerPoolIndex = _templeIndexInPool;
+        balancerPoolId = _balancerPoolId;
+    }
+
+    function setPoolHelper(address _poolHelper) external onlyOwner {
+        poolHelper = AMO__IPoolHelper(_poolHelper);
+
+        emit SetPoolHelper(_poolHelper);
+    }
+
+    function setPostRebalanceSlippage(uint64 slippage) external onlyOwner {
+        if (slippage >= TPF_PRECISION || slippage == 0) {
```

Comment:
Should TPF_PRECISION be called SLIPPAGE_PRECISION (or just BPS_PRECISION for basis points)